usr/sbin/p11sak/p11sak.c

```diff
@@ -1141,6 +1191,200 @@ static CK_ATTRIBUTE_TYPE col2type(int col)
     }
 }
 
+/**
+ *  Print in p11sak_defined_attrs.conf defined attributes in short format
+ */
+static CK_RV short_print_vendor() 
+{
+    CK_RV rc = CKR_OK;
+    int f;
+    FILE *fp;
+    struct ConfigBaseNode *cfg, *c, *name, *hex_string, *type;
+    struct ConfigStructNode *structnode;
+    char file_loc[1024] = P11SAK_DEFINED_ATTRS_LOCATION;
+
+    // get file location
+    cfg = NULL;
+    if (strcmp(P11SAK_DEFINED_ATTRS_LOCATION, "") == 1) {
+        strcpy(file_loc, P11SAK_DEFINED_ATTRS_LOCATION);
+    } else {
+        strcpy(file_loc, P11SAK_DEFAULT_CONF_FILE);
+    }
+
+    // open and parse file
+    if ((fp = fopen(file_loc, "r")) == NULL) {
+        fprintf(stderr, "\nFailed to load %s: %s\n", file_loc, strerror(errno));
+        return CKR_ARGUMENTS_BAD;
+    }
+    if (parse_configlib_file(fp, &cfg, error_hook, 0)) {
+        fprintf(stderr, "\nFailed to parse %s\n", file_loc);
+        return CKR_DATA_INVALID;
```

Comment:
Here you leak `fp` since you never close it.  Easier way is to store the result of `parse_configlib_file` in a variable, `fclose(fp);` and then evaluate the return value of `parse_configlib_file`.

Reply:
fp leak fixed